jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/skaffold/FilesMojo.java

```diff
@@ -0,0 +1,102 @@
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.maven.skaffold;
+
+import com.google.cloud.tools.jib.maven.MavenProjectProperties;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableSet;
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.model.FileSet;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+
+/**
+ * Print out changing source dependencies on a module. In multimodule applications it should be run
+ * by activating a single module and it's dependent modules. Dependency collection will ignore
+ * project level snapshots (sub-modules) unless the user has explicitly installed them (by only
+ * requiring dependencyCollection). For use only within skaffold.
+ *
+ * <p>Expected use: "./mvnw jib:_skaffold-files -q" or "./mvnw jib:_skaffold-files -pl module -am
+ * -q"
+ */
+@Mojo(
+    name = FilesMojo.GOAL_NAME,
+    requiresDependencyCollection = ResolutionScope.COMPILE_PLUS_RUNTIME)
+public class FilesMojo extends AbstractMojo {
+
+  @VisibleForTesting static final String GOAL_NAME = "_skaffold-files";
+
+  @Nullable
+  @Parameter(defaultValue = "${project}", readonly = true)
+  private MavenProject project;
+
+  // This parameter is cloned from JibPluginConfiguration
+  @Nullable
+  @Parameter(defaultValue = "${project.basedir}/src/main/jib", required = true)
+  private File extraDirectory;
+
+  @Override
+  public void execute() throws MojoExecutionException, MojoFailureException {
+    Preconditions.checkNotNull(project);
+    Preconditions.checkNotNull(extraDirectory);
+
+    // print out pom configuration files
+    System.out.println(project.getFile());
+    if ("pom".equals(project.getPackaging())) {
+      // done if <packaging>pom</packaging>
+      return;
+    }
+
+    // print out sources directory
+    System.out.println(project.getBuild().getSourceDirectory());
+
+    // print out resources directory
+    ImmutableSet.copyOf(project.getBuild().getResources())
+        .stream()
+        .map(FileSet::getDirectory)
+        .forEach(System.out::println);
+
+    // this seems weird, but we will only print out the "jib" directory on projects where
+    // the plugin is explicitly configured (even though _skaffold-files is a jib-maven-plugin goal).
+    if (project.getPlugin(MavenProjectProperties.PLUGIN_KEY) != null) {
+      // print out extra directory
+      System.out.println(extraDirectory.getAbsoluteFile().toPath());
```

Comment:
Perhaps we should have a method that always prints the absolute path and use that method to make sure all the prints print out an absolute path? Like:

```java
private static void printFile(Path file) {
  System.out.println(file.toAbsolutePath());
}
```

Reply:
I mean it's a question of ensuring we using printFile vs file.toAbsolutePath, but there's no way to enforce we're actually calling either. I'd rather not someone reading this have to go find out what "printFile" is doing. Absolute file path printing is ensure in the tests at the moment.